logging/dw_es_domain.yaml

```diff
@@ -206,14 +131,70 @@ Resources:
-                                  - !If [ ValidNotAnythingWhitelistCIDR2, !Ref "WhitelistCIDR2", !Ref "AWS::NoValue" ]
-                                  - !If [ ValidNotAnythingWhitelistCIDR3, !Ref "WhitelistCIDR3", !Ref "AWS::NoValue" ]
+                                  - !If [ HasWhitelistCIDR1, !Ref "WhitelistCIDR1", !Ref "AWS::NoValue" ]
+                                  - !If [ HasWhitelistCIDR2, !Ref "WhitelistCIDR2", !Ref "AWS::NoValue" ]
+                                  - !If [ HasWhitelistCIDR3, !Ref "WhitelistCIDR3", !Ref "AWS::NoValue" ]
             AdvancedOptions:
                 rest.action.multi.allow_explicit_index: true
+                # See doc: indices.fielddata.cache.size: 40
+            CognitoOptions:
+                # We are only using IP restrictions for now.
+                Enabled: false
+            DomainName: !Ref DomainName
+            EBSOptions:
+                EBSEnabled: true
+                Iops: 0
+                VolumeSize: !Ref NodeStorageSize
+                VolumeType: "gp2"
+            ElasticsearchClusterConfig:
+                DedicatedMasterEnabled: false
+                InstanceCount: !Ref NodeCount
+                InstanceType: !Ref InstanceType
+                ZoneAwarenessEnabled: false
+            ElasticsearchVersion: !Ref ElasticsearchVersion
+            EncryptionAtRestOptions:
+                Enabled: true
+                KmsKeyId: !Sub "arn:aws:kms:${AWS::Region}:${AWS::AccountId}:alias/aws/es"
+            NodeToNodeEncryptionOptions:
+                Enabled: true
+            SnapshotOptions:
+                AutomatedSnapshotStartHour: 22
             Tags:
                 - Key: user:sub-project
                   Value: logging-es-cluster
+        UpdatePolicy:
+            EnableVersionUpgrade: true
+
+    # Topic and alerts about the ES cluster itself.
+
+    ClusterAlertingTopic:
+         Type: AWS::SNS::Topic
+         Properties:
+             DisplayName: "ES alerts"
+             Tags:
+                 - Key: user:sub-project
+                   Value: logging-es-cluster
+
+    ClusterFreeStorageSpaceTooLowAlarm:
+        Type: 'AWS::CloudWatch::Alarm'
+        Properties:
+            AlarmActions:
+                - !Ref ClusterAlertingTopic
+            AlarmDescription: "Cluster is running low on free storage."
+            ComparisonOperator: LessThanThreshold
+            Dimensions:
+                -   Name: ClientId
+                    Value: !Ref "AWS::AccountId"
+                -   Name: DomainName
+                    Value: !Ref ElasticsearchDomain
+            EvaluationPeriods: 1
+            MetricName: FreeStorageSpace
+            Namespace: AWS/ES
+            OKActions:
+                - !Ref ClusterAlertingTopic
+            Period: 60
+            Statistic: Minimum
+            Threshold: 2000  # 2 Gigabyte in Megabyte
```

Comment:
2 gigs will be enough? or we have a TTL policy somewhere?

Reply:
The thinking is that with 20GiB in the cluster, that's 10%. So as long as someone starts deleting old indices within, say, a week, this is ok.